### ISSUE TITLE
Drop enums in favour of free text

### ISSUE THREAD

**Issue by milstan** [uropa-project/uropa#10]
Most of the properties allow for usage of flexible categories (e.g. `dataCategories.name` is a string), however `legalBasisType` is an enum.

For easier use in conjunction with [PRIV](https://github.com/blindnet-io/product-management/blob/main/refs/schemas/priv/RFC-PRIV.md) it would be better to drop all the enums, and leave it to the implementing system to adopt its own categories.

PRIV relies of globally-defined categories to enable [automation](https://github.com/blindnet-io/devrel-management/blob/main/docs/use-cases/generic.md). We do have a mapping of our categories to UROPA enums, but it would be more convenient if UROPA didn't have any enums, so that native PRIV terms can be used insted.

**Comment by Jabuf** [comment]
Hello @milstan, thanks for the feedback.
There's two main reasons we choose to go for enums and that it's unlikely to changes.

The first, is that having values to choose ease the user experience. Indeed if someone with a limited knowledge in ROPA has to work on one it will be easier for him to check the possible values and choose one from it and will ensure a more accurate filling of the ROPA.

The second, and by far the most important one, is that having enums will allow the format to be more universal. In fact without that the tools that you could build on that format would be very limited.
For example let's say that we change `legalBasisType` to a string. Then the tools of `PRIV` will generate Uropa files using their own values and let's imagine another company doing the same. That would means that now if someone want to build a software using Uropa files it will be impossible for it to do automatic process using the value of this field since there's no common reference.
In short we're using enums on `Uropa` probably for the same reason you're using enums on `PRIV`.

**Comment by milstan** [comment]
Thanks for your answer.

I understand your reason for using enums. Systems that exchange data need shared semantics. Of course.

My suggestion was to separate the format from the enum vocabulary.

Systems that exchange data anyway need to agree on a common format (and vocabulary).

That would increase the usability of UROPA, because we could use UROPA format, and not EROPA enums. 

In the concrete case, the UROPA enums are not expressive enough for the automation we want to achieve with PRIV.

By separating enums vocabulary from the format itself you can have more people using your format. We would in fact consider using it. But it needs to be compatible with PRIV terms - otherwise it would not be useful for us at all.

**Comment by Jabuf** [comment]
Our goal ultimately is to have `Uropa` files being able to be used by **universal** tools. If we don't use enums on relevant fields, it won't be possible, or very limited.
Having enums would also ensure that an Uropa file can be used by all softwares compatible with Uropa. 

For example, there's this [website](https://uropa-ui.cleverapps.io/) that allow you to fill an `Uropa` file. Imagine a company that would have filled all of its ROPAs using this website. Now let's say that they want to use a tool (something similar to what you're doing). It would be a shame if it wasn't possible because the values of some fields are incompatible, while having enums would ensure that it won't be the case.

Overall having a format where files won't have a shared semantic, would greatly diminish its purpose. Yes it means that existing tools will need a additional layer of abstraction to map their values with the ones of `Uropa`. But that's precisely one of the goal of `Uropa`, having a common reference for these values.

On another hand if you think that some of our enums aren't precise enough you are more than welcome to make suggestions.

**Comment by milstan** [comment]
> Our goal ultimately is to have Uropa files being able to be used by universal tools. 

@Jabuf thank you. I perfectly understand your goal. My understanding is however that your current design goes priceisely against this goal. That is why I made the comment I made.

The current desing means that systems needing more expressive vocabularies (we for isntance need a shared vocabulary for not only legal bases, but also Data Categories, Purposes, etc.) can't use UROPA.

On the other hand if you split UROPA in two parts:
A - One only focused on the format of data, and
B - A vocabulary of enums

People might be able to use your format (A) and extend your vocabulary of enums (or use an alternative one).

If you only recommend using your enums, instead of making them mandatory you'd increase the likelyhood that people use your format (part A). 

There would be a clear gain in interoperability. Two systems using the same format (A) and using different vocabularies can still interoperate - there is agreat deal of metadata that is covered by the format itself. Systems using alternative (or extended) vocabularies of enums (B) are then responsable for implementing mappings to your vocabulary of enums, and can still interoperate with other systems using your core vocabulary.

But I understand we don't have the same vision on optimal model reutilisability. So I won't insist :)

**Comment by Jabuf** [comment]
> The current desing means that systems needing more expressive vocabularies (we for isntance need a shared vocabulary for not only legal bases, but also Data Categories, Purposes, etc.) can't use UROPA.

We've tried to use enums when it make sense, i.e. when the possible values are predefined. 
For example a legal basis must belong to one of the six defined by the [ chapter 6 of the GDPR](https://gdpr-info.eu/art-6-gdpr/). In this case the pros of using an enum far outweight the cons, and honestly it's not even close (I could explain a more if you wish). That's why we won't drop enums as a whole, since they are needed on some fields.

However I can see that not being the case for all enums. This is primarily a community project, if you think some fields should or shouldn't be enums we're open to suggestions. The same goes if you think some of our current enums aren't precise enough. If you do so I'll appreciate separate issues for each concerned fields, it'll help us tracking them more efficiently.

**Comment by milstan** [comment]
I understand. Indeed it is a community project, so this was my suggestion that I see as possible improvement. It's not compatible with your design choices, and that's OK. I'm closing this issue.
